storage/storage-client/src/lib.rs

```diff
@@ -54,6 +55,63 @@ use storage_proto::{
 };
 use tokio::runtime::Runtime;
 
+pub struct StorageServiceClientSimple {
+    network_client: Mutex<NetworkClient>,
+}
+
+impl StorageServiceClientSimple {
+    pub fn new(server_address: &SocketAddr) -> Self {
+        Self {
+            network_client: Mutex::new(NetworkClient::new(*server_address)),
+        }
+    }
+
+    fn request(
+        &self,
+        input: storage_interface::StorageMsg,
+    ) -> Result<Vec<u8>, storage_interface::Error> {
+        let input_message = lcs::to_bytes(&input)?;
+        let mut client = self.network_client.lock().unwrap();
+        client.write(&input_message)?;
+        let result = client.read()?;
+        Ok(result)
+    }
+
+    pub fn get_account_state_with_proof_by_version(
+        &self,
+        address: AccountAddress,
+        version: Version,
+    ) -> Result<(Option<AccountStateBlob>, SparseMerkleProof), storage_interface::Error> {
+        let response = self.request(
+            storage_interface::StorageMsg::GetAccountStateWithProofByVersionRequest(Box::new(
+                storage_interface::GetAccountStateWithProofByVersionRequest::new(address, version),
+            )),
+        )?;
+        lcs::from_bytes(&response)?
+    }
+
+    pub fn get_startup_info(&self) -> Result<Option<StartupInfo>, storage_interface::Error> {
+        let response = self.request(storage_interface::StorageMsg::GetStartupInfoRequest)?;
+        lcs::from_bytes(&response)?
+    }
+
+    pub fn save_transactions(
+        &self,
+        txns_to_commit: Vec<TransactionToCommit>,
+        first_version: Version,
+        ledger_info_with_sigs: Option<LedgerInfoWithSignatures>,
+    ) -> Result<(), storage_interface::Error> {
+        let response = self.request(storage_interface::StorageMsg::SaveTransactionsRequest(
+            Box::new(storage_interface::SaveTransactionsRequest::new(
+                txns_to_commit,
+                first_version,
+                ledger_info_with_sigs,
+            )),
+        ))?;
+        lcs::from_bytes(&response)?
+    }
+}
+
 /// This provides storage read interfaces backed by real storage service.
 pub struct StorageReadServiceClient {
```

Comment:
I see why you named it as such... In general this naming is kinda interesting.. a client /needs/ a service to speak to. So the word service here is really unnecessary.

Reply:
lol. Yeah, but let's keep it as it is.